src/scripts/modules/add_destination.js

```diff
@@ -0,0 +1,130 @@
+import destinationsList from "../destinationList.js";
+import events from "./pubsub.js";
+
+let form;
+let uploadBtn;
+let radioBtns;
+let radioVisited;
+let radioBucketlist;
+let cityInput;
+let countryInput;
+let photoInput;
+let newCity;
+let newCountry;
+let newPhoto;
+// let updatedList
+
+function cacheDom() {
+  form = document.getElementById("add-city-form");
+  uploadBtn = document.getElementById("upload");
+  radioBtns = document.querySelectorAll("input[type=radio]");
+  radioVisited = document.getElementById("rd-visited");
+  radioBucketlist = document.getElementById("rd-bucketlist");
+  cityInput = document.getElementById("inputCity");
+  countryInput = document.getElementById("inputCountry");
+  photoInput = document.getElementById("inputImg");
+}
+
+function bindEvents() {
+  form.addEventListener("submit", (event) => {
+    setValues();
+    addDestination();
+    event.preventDefault();
+    console.log(destinationsList);
+  });
+
+  // events.publish('destinationAdded', (form.addEventListener("submit", (event) => {
+  //   setValues();
+  //   addDestination();
+  //   event.preventDefault();
+  //   console.log(destinationsList);
+  // })));
+
+  // updatedList = form.addEventListener("submit", (event) => {
+  //   setValues();
+  //   addDestination();
+  //   event.preventDefault();
+  //   console.log(destinationsList);
+  //   events.publish('destinationAdded', updatedList)
+  // });
+
+  radioVisited.addEventListener("click", isVisited);
+  radioBucketlist.addEventListener("click", isBucketlist);
+  cityInput.addEventListener("change", (event) => {
+    updateCity("city", event.target.value);
+  });
+  countryInput.addEventListener("change", (event) => {
+    updateCountry("country", event.target.value);
+  });
+  photoInput.addEventListener("change", (event) => {
+    updatePhoto("photo", event.target.value);
+  });
+}
+
+function setValues() {
+  return {
+    city: cityInput,
+    country: countryInput,
+    visited: "",
+    url: photoInput,
+  };
+}
+
+function addDestination() {
+  return destinationsList.push({
+    ...newCity,
+    ...newCountry,
+    ...newPhoto,
+  });
+}
+function isVisited() {
+  if (radioVisited.checked === true) {
+    newCity.visited = true;
+  }
+}
+
+function isBucketlist() {
+  if (radioBucketlist.checked === true) {
+    newCity.visited = false;
```

Comment:
Continuing my comment about `formState`, here and in `isVisited`, you can also do:

```js
updateFormState('visited', false)
```